Add unit selection to results page

lboeman opened the pull request:
Adds another select option if the variable is a power or irradiance value as seen below. If the user selects a new unit and clicks "add to plot" again, the units are updated. 
![Screenshot from 2021-03-16 14-21-39](https://user-images.githubusercontent.com/21206164/111381621-2e345d00-8663-11eb-9411-50f0edf5f2b1.png)

alorenzo175 reviewed (approved):
Cool. And a selector and conversion for the table will come later?

lboeman commented:
> Cool. And a selector and conversion for the table will come later?

Ah, yep thanks I'll add that to this PR.

lboeman commented:
Added summary table unit selection as well, and updated the formatter to round to allow for precision to the nearest watt instead of always displaying an integer value.. Is that reasonable?

alorenzo175 commented on dashboard/src/utils/unitConversion.ts:
```diff
@@ -0,0 +1,53 @@
+  GWh: 1000000000
+};
+
+const insolationUnitValues: Record<string, number> = {
```
```suggestion
const irradianceUnitValues: Record<string, number> = {
```

alorenzo175 commented on dashboard/src/utils/unitConversion.ts:
```diff
@@ -0,0 +1,53 @@
+  "MW/m^2": 1000000,
+  "GW/m^2": 1000000000
+};
+
```
```suggestion

const insolationUnitValues: Record<string, number> = {
  "Wh/m^2": 1,
  "kWh/m^2": 1000,
  "MWh/m^2": 1000000,
  "GWh/m^2": 1000000000
};
```

alorenzo175 commented on dashboard/src/utils/unitConversion.ts:
```diff
@@ -0,0 +1,53 @@
+  } else if (units in energyUnitValues) {
+    return energyUnitValues;
+  } else if (units in insolationUnitValues) {
+    return insolationUnitValues;
```
```suggestion
    return insolationUnitValues;
  } else if (units in irradianceUnitValues) {
    return irradianceUnitValues;
```

alorenzo175 commented on dashboard/src/components/jobs/CustomResultPlots.vue:
```diff
@@ -200,14 +228,27 @@ export default class CustomPlots extends Vue {
+    for (const index in this.selectedVariables) {
+      const variable = this.selectedVariables[index];
+      // @ts-expect-error
+      convertable[index] = ["W", "Wh", "W/m^2"].includes(
```
```suggestion
      convertable[index] = ["W", "Wh", "W/m^2", "Wh/m^2"].includes(
```

alorenzo175 reviewed (commented):
I would actually keep everything as integers. Usually when your changing units like that it's because those extra digits don't matter. Maybe @wholmgren has an opinion on that

alorenzo175 commented:
very nice!
